# exo1
Using escape sequence print each word in different lines. Sample:

In [12]:

s = "hey! welcome to edureka"
s1 = s.split(" ")

for x in s1:
    print(x)

hey!
welcome
to
edureka


# exo2
Write a program to create a list of even numbers starting from x to y.

In [14]:
x, y = input().split(' ')
print(x, y)
l = list(range(int(x), int(y), 2))
print(l)


100 201
[100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200]


# exo3


In [20]:
st = input() 
x, y = input().split(' ')
print(st)
st = st.replace(x, y)
print(st)

i love apples 
i prefer apples 


# exo4


In [36]:
N = int(input("where do you want to stop? :"))
u0 = 0
u1 = 1
print(u0)
print(u1)
for i in range(2, N):
    un = u0 + u1
    u0 = u1 
    u1 = un
    print(un)


0
1
1
2
3
5
8
13
21
34


# exo5

In [46]:
num = int(input())
s = 0
while(num != 0):
    s = s + num % 10
    num = num // 10
print(s)



21


# exo6
Write a program to find factorial of a given number N.

In [22]:
num = int(input())
f  = num
while(num >= 2):
    num = num - 1
    f = f * num
print(f)

120


# exo7
Write a program to generate the following pattern:
\n*
\n**
\n***
\n****

In [25]:
for i in range(6):
    print('*'*i)


*
**
***
****
*****


# exo8
Write a program to generate the following pattern.

In [31]:
j = 4
for i in range(1,10,2):
    print(' '*j+'*'*i)
    j -= 1


    *
   ***
  *****
 *******
*********


# exo9
Write a program to generate the following pattern for given depth N.

In [47]:
N  = int(input())
j = 1
for i in range(N, 0, -1):
    print(' '*i+'*'*j)
    j += 2
    

          *
         ***
        *****
       *******
      *********
     ***********
    *************
   ***************
  *****************
 *******************


# exo10
Write a program to generate the following output.

In [59]:
for i in range(1, 6):
    for j in range(i, i*2):
        print(j, end=' ')
    print()

1 
2 3 
3 4 5 
4 5 6 7 
5 6 7 8 9 


# exo11
Write a program that accepts N values and print it as a list

In [68]:
l = []
N = int(input())
for i in range(N):
    l.append(int(input()))
print(l)

[1, 2, 3, 4, 5]


# exo12
Write a program to find the given number is Armstrong number or not and print out as 'Yes' or 'No'.
[Hint: Armstrong Number is sum of its own digits raised to power of 3]
Input:153
Output:Yes

method 1:

In [76]:
rest = 0
s = 0
N = int(input())
Nt = N
while(N != 0):
    rest = N % 10
    s = s + rest **3
    N  = N // 10
if s == Nt:
    print('Yes')
else:
    print('No')


No


method 2:

In [92]:
s = 0
n = input()
for i in list(n):
    s = s + int(i)**3
if s == int(n):
    print('Yes')
else:
    print('No')


Yes


# exo13
Write a program to find the N th number of a Arthematic Progression when initial number(a) and
difference( d ) are given.
Input:
The input is N, a, d each in newline
4
2
3
Output:
11

In [97]:
N = int(input())
a = int(input())
d = int(input())
print(a + (N-1)*d)

 

11


# exo14
Write a program to generate a sequence of N numbers of a geometric progression when intial number ( a )
and common ratio( r ) is given. print the sequence as a list.
Input:
The input is N,a,r seperated by comma.
5,2,3
Output: [2, 6, 18, 54, 162]

method 1

In [103]:
N = int(input())
a = int(input())
r = int(input())
l = [] 
for i in range(N):
    l.append(a)
    a = a*r
print(l)


[2, 6, 18, 54, 162]


method 2

In [106]:
N, a, r = map(int, input().split(' '))
l = []
for i in range(N):
    l.append(a*r**i)
print(l)

[2, 6, 18, 54, 162]


# exo15
Write a program to find the count of each aplhabet in the given sentence.

In [116]:
sentence = input()
set_sentence = set(sentence)
for i in set_sentence:
    print(i,sentence.count(i))

i 2
e 1
p 1
  2
l 1
f 2
o 1


# exo16 
Write a program that calculate number of lowercase, uppercase and numbers alphabets.

In [122]:
low = 0
up = 0
num = 0
sentence = input()
print(sentence)
for i in sentence:
    if i.islower():
        low = low + 1
    elif i.isupper():
        up = up + 1
    elif i.isdigit():
        num = num + 1
print('Uppercase characters: ', up)
print('Lowercase characters: ', low)
print('Numbers: ', num)

Happy Birthday Draven! Now you are 26.
Uppercase characters:  4
Lowercase characters:  24
Numbers:  2


# exo17
Write a program to find whether a given string is palindrome or not

In [133]:
word = input()
if word == word[::-1] :
    print('Yes')
else:
    print('No')

Yes


# exo18
Write a program that accepts a sequence of numbers and print the numbers after sorting them seperated
by comma.

In [148]:
N = int(input())
l = []
for i in range(N):
    l.append(int(input()))
l.sort()
print(l)


[0, 3, 5, 7, 8]


# exo19
Given below a list of integers, find the triplets whose sum if equal to zero and print them as tuples.

In [166]:
int_list = [0, 2, 3, 4, 5, 6, 9, -1, -9, -7, -6, -5, -4, -3, -2]
l = len(int_list)
tuplelist = []
t = []
for i in range(l-2):
    for j in range(i+1,l-1):
        for k in range(j+1, l):
            if int_list[i]+int_list[j]+int_list[k] == 0:
                tuplelist.append(tuple([int_list[i],int_list[j],int_list[k]]))
for i in tuplelist:
    print(i)






(0, 2, -2)
(0, 3, -3)
(0, 4, -4)
(0, 5, -5)
(0, 6, -6)
(0, 9, -9)
(2, 3, -5)
(2, 4, -6)
(2, 5, -7)
(3, 4, -7)
(3, 6, -9)
(3, -1, -2)
(4, 5, -9)
(4, -1, -3)
(5, -1, -4)
(5, -3, -2)
(6, -1, -5)
(6, -4, -2)
(9, -7, -2)
(9, -6, -3)
(9, -5, -4)
[(0, 2, -2), (0, 3, -3), (0, 4, -4), (0, 5, -5), (0, 6, -6), (0, 9, -9), (2, 3, -5), (2, 4, -6), (2, 5, -7), (3, 4, -7), (3, 6, -9), (3, -1, -2), (4, 5, -9), (4, -1, -3), (5, -1, -4), (5, -3, -2), (6, -1, -5), (6, -4, -2), (9, -7, -2), (9, -6, -3), (9, -5, -4)]


# exo 20
Using for loop print aplhabets from a-z seperated by space.

In [156]:

for i in range(97, 123):
    print(chr(i), end = ' ')

a b c d e f g h i j k l m n o p q r s t u v w x y z 

# exo22
Write a program which accepts a string from console and print it in reverse order.

In [167]:
sentence = input()
print(sentence[::-1])

yek epacsE eht no regnif eno peek syawlA


# exo23
Write a program which accepts a string from console and print the words in reverse order.

In [185]:
sentence = input().split(' ')
for word in sentence:
    print(word[::-1], end = ' ')
    

syawlA peek eno regnif no eht epacsE yek 

# exo24
Write a program to calculate the numbers between X and Y which are divisible by 5 and 13 but not divisible
by 10 seperated by space.

In [198]:
x, y = map(int, input().split(' '))
for i in range(x, y+1):
    if (i % 5 == 0) and (i % 13 == 0) and (i % 10 != 0):

        print(i, end = ' ')

65 195 

# exo25
Write a program to calculate all possible permutations for a given sequence of numbers.
Input: N is length of sequence and next line contains sequence seperated by space.
3
4,6,2

In [225]:
N = int(input())
l = []
sequence = []
for i in range(N):
    l.append(int(input()))
for i in range(N):
    for j in range(N):
        if j != i:
            for k in range(N):
                if k != i and k != j:
                    inner_sequence = []
                    inner_sequence.extend([l[i], l[j], l[k]])
                    sequence.append(inner_sequence)
print(sequence) 
    

[[2, 4, 6], [2, 6, 4], [4, 2, 6], [4, 6, 2], [6, 2, 4], [6, 4, 2]]


# exo26
Samantha represents the performance of students using dictionary in tests as follows:
{'English':{'Sam':60,'Jackson':74}, 'History':{'Gloria':83,'Sam':35} ,....}
Your task is to create a dictionary with student names as keys and their average marks from all subjects.
Consider the below dictionary as input

In [298]:
tests = {'English':{'Sam':60,'Jackson':74,'Ahree':85},
'History':{'Gloria':83,'Sam':65,'Isla':78,'Aron':72,'Gray':61},
'Geography':{'Jackson':92,'Gloria':95,'Isla':82,'Aron':75,'Ahree':76},
'Mathematics':{'Sam':99,'Gloria':74,'Jackson':89,'Ahree':85,'Gray':95},
'Science':{'Sam':89,'Aron':82,'Gray':78,'Isla':93,'Ahree':87}
}
l = []
for i in tests:
    l.extend(tests[i].keys())
l = set(l)
dic = {}
dic = dict.fromkeys(l, 0)

for i in tests:
    for key in tests[i]:
        dic[key] = dic[key]+tests[i][key]
for key in dic:
    dic[key] = dic[key]/5
print(dic)



{'Isla': 50.6, 'Aron': 45.8, 'Ahree': 66.6, 'Sam': 62.6, 'Jackson': 51.0, 'Gray': 46.8, 'Gloria': 50.4}


# exo27
Create a dictionary from the following keys: 'Name', 'Class', 'Age'
with empty lists and input the data of five students

In [242]:
dic = {'Name': [], 'Class': [], 'Age': []}
for i in range(2):
    dic['Name'].append(input())
    dic['Class'].append(input())
    dic['Age'].append(input())
print(dic)

{'Name': ['aze', 'rty'], 'Class': ['qsd', 'fgh'], 'Age': ['wxc', 'cvb']}


# exo28
Write a program to concatenate the following dictionaries:
dict1={'A':10, 'B':20}
dict2={'C':30, 'D':40}


In [246]:
dict1={'A':10, 'B':20}
dict2={'C':30, 'D':40}
dict3={'E':50,'F':60}
dict = {}
dict.update(dict1)
dict.update(dict2)
dict.update(dict3)
print(dict)

{'A': 10, 'B': 20, 'C': 30, 'D': 40, 'E': 50, 'F': 60}


# exo29
Write a program to generate prime numbers from 1 to 1000 and save them to a text file. Read N lines of the
previous file.

In [311]:
obj_file = open("exo29","w")
prim = True
for i in range(2, 1000):
    prim = True
    for j in range(2, i//2+1):
        if(i % j == 0):
            prim = False
    if prim == True:
        obj_file.write(str(i)+' ')
obj_file.close()

obj_file = open("exo29","r")
for i in obj_file:
    print(i)
obj_file.close()

            


2 3 5 7 11 13 17 19 23 29 31 37 41 43 47 53 59 61 67 71 73 79 83 89 97 101 103 107 109 113 127 131 137 139 149 151 157 163 167 173 179 181 191 193 197 199 211 223 227 229 233 239 241 251 257 263 269 271 277 281 283 293 307 311 313 317 331 337 347 349 353 359 367 373 379 383 389 397 401 409 419 421 431 433 439 443 449 457 461 463 467 479 487 491 499 503 509 521 523 541 547 557 563 569 571 577 587 593 599 601 607 613 617 619 631 641 643 647 653 659 661 673 677 683 691 701 709 719 727 733 739 743 751 757 761 769 773 787 797 809 811 821 823 827 829 839 853 857 859 863 877 881 883 887 907 911 919 929 937 941 947 953 967 971 977 983 991 997 


# exo30
Create a game of Rock, Papers and Scissors against computer using Python. Rules for the game are given
as follows:
Rock beats Scissor
Scissor beats Paper
Paper beats Rock
If the play loses ask him either he wants to payback or not. If the response is 'Yes' then continue the game.
If the player says 'No' then display the final scores. Input moves as R for Rock, P for Paper and S for
scissors. Response to every move should be proper example:

In [323]:
import random
play =1
while(play == 1):
    print("R: Rock")
    print("P: Paper")
    print("S: Scissors")
    print("Choose your role: ")
    p = input()
    c = random.choice(["R", "S", "P"])
    print(c)
    if ((p=="R" and c== "S" ) or (p=="S" and c=="P") or ( p =="P" and c=="R")):
        print("Win")
    else:
        print("Lose")
    print("Play again? (1 for Yes /0 for No)")
    play = input()
    if( play == 0):
        print("Good bye")




R: Rock
P: Paper
S: Scissors
Choose your role: 
S
Lose
Play again? (1 for Yes /0 for No)


In [4]:
tables = [lambda x: x*10 for x in range(1, 11)]
 
for table in tables:
    print(table())

TypeError: <lambda>() missing 1 required positional argument: 'x'